db/db_impl_open.cc

```diff
@@ -347,6 +347,13 @@ Status DBImpl::Recover(
           return s;
         }
       } else {
+        env_->DeleteFile(LockFileName(dbname_));
+        std::string db_absolute_path;
+        env_->GetAbsolutePath(dbname_, &db_absolute_path);
+        std::string logfname =
+            InfoLogFileName(dbname_, db_absolute_path, immutable_db_options_.db_log_dir);
+        env_->DeleteFile(logfname);
```

Comment:
I don't think we should delete the log file unless we're sure this call to ```DB::Open()``` created it.

Reply:
Thanks. Taken care of this by keeping track of created directories and cleaning up on failure.